src/graphql/types/object/btc-wallet.ts

```diff
@@ -43,8 +46,26 @@ const BtcWallet = GT.Object<Wallet>({
     },
     transactions: {
       type: TransactionConnection,
-      args: connectionArgs,
+      args: { ...connectionArgs, addresses: { type: GT.List(OnChainAddress) } },
       resolve: async (source, args) => {
+        let { addresses } = args
+        if (addresses) {
+          addresses = addresses.length ? addresses : source.onChainAddresses()
+          for (const address of addresses) {
+            if (address instanceof InputValidationError) throw address
```

Comment:
shouldn't be just `Error` ?

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1511/commits/c7e25002ea2758ee3266f4133cf82eebb2202ffc